src/api/app/controllers/staging/workflows_controller.rb

```diff
@@ -20,22 +24,50 @@ def create
   end
 
   def destroy
-    staging_workflow = @project.staging
-    unless staging_workflow
+    authorize @staging_workflow
+
+    @staging_workflow.destroy!
+  end
+
+  def accept_staging_projects
+    authorize @staging_workflow, :update?
+
+    projects_to_accept = @staging_workflow.staging_projects.where(name: @staging_projects)
+    missing_projects = @staging_projects - projects_to_accept.pluck(:name)
+    if missing_projects.present?
       render_error(
-        status: 404,
-        errorcode: 'not_found',
-        message: "Project '#{@project}' doesn't have staging workflow."
+        status: 400,
+        errorcode: 'invalid_request',
+        message: "API request contains projects that are not part of the workflow: '#{missing_projects.join(', ')}'"
+      )
+      return
+    end
+    projects_not_acceptable = projects_to_accept.reject { |project| project.overall_state == :acceptable }
+    if projects_not_acceptable.present?
+      render_error(
+        status: 400,
+        errorcode: 'invalid_request',
+        message: "API request contains projects that are not in state acceptable: '#{projects_not_acceptable.join(', ')}'"
       )
       return
     end
-    authorize staging_workflow
 
-    staging_workflow.destroy!
+    projects_to_accept.each { |project| StagingProjectAcceptJob.perform_now(project.id, User.current.login) }
```

Comment:
What's the difference between a job performed blocking and a service module? I find this rather strange.